joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -399,13 +403,70 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
+      }
+      .foreach { case (name, fullName) => callNode_.name(name).methodFullName(fullName) }
+    callAst(callNode_, quasisArg +: expressionArgs, receiver = Option(receiverAst), base = Option(Ast(baseNode)))
+  }
+
+  /** Lowering from expressions like, x`a ${1+1} b` to x(["a ", " b"], 1+1)
+    */
   def astForTemplateExpression(templateExpr: BabelNodeInfo): Ast = {
-    val argumentAst      = astForNodeWithFunctionReference(templateExpr.json("quasi"))
-    val callName         = code(templateExpr.json("tag"))
-    val callCode         = s"$callName(${codeOf(argumentAst.nodes.head)})"
-    val templateExprCall = callNode(templateExpr, callCode, callName, DispatchTypes.STATIC_DISPATCH)
-    val argAsts          = List(argumentAst)
-    callAst(templateExprCall, argAsts)
+    val callee     = createBabelNodeInfo(templateExpr.json("tag"))
+    val calleeCode = callee.code
+    val (receiverAst, baseNode, callName) = callee.node match {
+      case MemberExpression =>
+        val base   = createBabelNodeInfo(callee.json("object"))
+        val member = createBabelNodeInfo(callee.json("property"))
+        base.node match {
+          case ThisExpression =>
+            val receiverAst = astForNodeWithFunctionReference(callee.json)
+            val baseNode    = identifierNode(base, base.code).dynamicTypeHintFullName(typeHintForThisExpression())
+            scope.addVariableReference(base.code, baseNode)
+            (receiverAst, baseNode, member.code)
+          case Identifier =>
+            val receiverAst = astForNodeWithFunctionReference(callee.json)
+            val baseNode    = identifierNode(base, base.code)
+            scope.addVariableReference(base.code, baseNode)
+            (receiverAst, baseNode, member.code)
+          case _ =>
+            val tmpVarName  = generateUnusedVariableName(usedVariableNames, "_tmp")
+            val baseTmpNode = identifierNode(base, tmpVarName)
+            scope.addVariableReference(tmpVarName, baseTmpNode)
+            val baseAst = astForNodeWithFunctionReference(base.json)
+            val code    = s"(${codeOf(baseTmpNode)} = ${base.code})"
+            val tmpAssignmentAst =
+              createAssignmentCallAst(Ast(baseTmpNode), baseAst, code, base.lineNumber, base.columnNumber)
+            val memberNode = createFieldIdentifierNode(member.code, member.lineNumber, member.columnNumber)
+            val fieldAccessAst =
+              createFieldAccessCallAst(tmpAssignmentAst, memberNode, callee.lineNumber, callee.columnNumber)
+            val thisTmpNode = identifierNode(callee, tmpVarName)
+            scope.addVariableReference(tmpVarName, thisTmpNode)
+
+            (fieldAccessAst, thisTmpNode, member.code)
+        }
+      case _ =>
+        val receiverAst = astForNodeWithFunctionReference(callee.json)
+        val thisNode    = identifierNode(callee, "this").dynamicTypeHintFullName(typeHintForThisExpression())
```

Comment:
I realize this is already done in `astForCallExpression`, and so I wouldn't change it for this PR. But I don't believe JS semantics ever call for an implicit `this` reference for the base. Under some circumstances the global object (e.g. `window`) will implicitly be used as the base / arg 0. But I think that's mostly a legacy thing in non-strict mode.

Reply:
W.r.t. code sharing and the legacy this: yes, I need to do a round of refactoring there but I want to have the basic functionality in to unblock you and the POC.